### PR TITLE
Make `add_item`,  `remove_item` and `clear_items` return self

### PR DESCRIPTION
**Pull request by HippoBaguette**
## Summary

This pull request makes `add_item`,  `return_item` and `clear_items` return `self`. This will allow method chaining
## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Rapptz** [comment]
I'm not sure how I feel about these being fluent. I think Embed's fluent design is already underused.

File: discord/ui/view.py [hunk 1]
```diff
@@ -279,8 +279,9 @@ def add_item(self, item: Item[Any]) -> None:
 
         item._view = self
         self.children.append(item)
+        return self
 
-    def remove_item(self, item: Item[Any]) -> None:
+    def remove_item(self, item: Item[Any]) -> View:
```
Comment (bijij) [inline discord/ui/view.py:284]:
```suggestion
    def remove_item(self, item: Item[Any]) -> Self:
```

File: discord/ui/view.py [hunk 2]
```diff
@@ -252,7 +252,7 @@ def from_message(cls, message: Message, /, *, timeout: Optional[float] = 180.0)
             view.add_item(_component_to_item(component))
         return view
 
-    def add_item(self, item: Item[Any]) -> None:
+    def add_item(self, item: Item[Any]) -> View:
```
Comment (bijij) [inline discord/ui/view.py:255]:
```suggestion
    def add_item(self, item: Item[Any]) -> Self:
```
We use `typing.Self` to denote when the self-type is being returned.

File: discord/ui/view.py [hunk 3]
```diff
@@ -295,11 +296,13 @@ def remove_item(self, item: Item[Any]) -> None:
             pass
         else:
             self.__weights.remove_item(item)
+        return self
 
-    def clear_items(self) -> None:
+    def clear_items(self) -> View:
```
Comment (bijij) [inline discord/ui/view.py:301]:
```suggestion
    def clear_items(self) -> Self:
```

**Review by bijij — CHANGES_REQUESTED** [review]

**Comment by HippoBaguette** [comment]
> I'm not sure how I feel about these being fluent. I think Embed's fluent design is already underused.

That's understandable, but is there really any harm in doing so?

File: discord/ui/view.py [hunk 4]
```diff
@@ -24,6 +24,7 @@
 
 from __future__ import annotations
 from typing import Any, Callable, ClassVar, Dict, Iterator, List, Optional, Sequence, TYPE_CHECKING, Tuple
+from typing_extensions import Self
```
Comment (bijij) [inline discord/ui/view.py:27]: Self is already imported in the `if TYPE_CHECKING:` block 
```suggestion
```

**Comment by HippoBaguette** [comment]
> There's no harm, but it'd have to be documented similarly to Embed.

I've updated the documentation to reflect it

File: discord/ui/view.py [hunk 5]
```diff
@@ -252,9 +252,12 @@ def from_message(cls, message: Message, /, *, timeout: Optional[float] = 180.0)
             view.add_item(_component_to_item(component))
         return view
 
-    def add_item(self, item: Item[Any]) -> None:
+    def add_item(self, item: Item[Any]) -> Self:
         """Adds an item to the view.
-
+
```
Comment (bijij) [inline discord/ui/view.py:257]:
```suggestion

```

Remove whitespace.

File: discord/ui/view.py [hunk 6]
```diff
@@ -252,9 +252,12 @@ def from_message(cls, message: Message, /, *, timeout: Optional[float] = 180.0)
...
         return view
 
-    def add_item(self, item: Item[Any]) -> None:
+    def add_item(self, item: Item[Any]) -> Self:
         """Adds an item to the view.
-
+        
+        This function returns the class instance to allow for fluent-style
+        chaining.
+
```
Comment (bijij) [inline discord/ui/view.py:260]:
```suggestion

```

File: discord/ui/view.py [hunk 7]
```diff
@@ -295,11 +302,17 @@ def remove_item(self, item: Item[Any]) -> None:
             pass
         else:
             self.__weights.remove_item(item)
+        return self
 
-    def clear_items(self) -> None:
-        """Removes all items from the view."""
+    def clear_items(self) -> Self:
+        """Removes all items from the view.
+
```
Comment (bijij) [inline discord/ui/view.py:309]:
```suggestion

```

File: discord/ui/view.py [hunk 8]
```diff
@@ -279,9 +282,13 @@ def add_item(self, item: Item[Any]) -> None:
 
         item._view = self
         self.children.append(item)
+        return self
 
-    def remove_item(self, item: Item[Any]) -> None:
+    def remove_item(self, item: Item[Any]) -> Self:
         """Removes an item from the view.
+
```
Comment (bijij) [inline discord/ui/view.py:289]:
```suggestion

```

**Review by bijij — CHANGES_REQUESTED** [review]
There shouldn't be any blank lines with whitespace, I suggest you run `black` on your code by installing black then typing `black discord` in a command line, or ensuring your editors formatting function uses black.

File: discord/ui/view.py (hunk 5 above)
Comment (HippoBaguette) [reply discord/ui/view.py:257]: > There shouldn't be any blank lines with whitespace, I suggest you run `black` on your code by installing black then typing `black discord` in a command line, or ensuring your editors formatting function uses black.

https://github.com/Rapptz/discord.py/blob/195c923becf0b691bf82786aa5773561c77cf716/discord/embeds.py#L352-L359

Looking in the embeds code there are white spaces

Comment (Rapptz) [reply discord/ui/view.py:257]: The problem is you have trailing whitespace which leads to it failing CI.

**Pull request merged**